evalml/pipelines/pipeline_base_meta.py

```diff
@@ -24,5 +25,9 @@ def _check_for_fit(self, X=None, y=None):
             elif y is None:
                 return method(self, X)
             else:
+                # For time series classification pipelines, predict will take X, y, objective
```

Comment:
The problem is that `TimeSeriesBinaryClassification.predict` accepts `X`, `y` and `objective`. All other pipeline classes accept `X` and `y` (`TimeSeriesRegressionPipeline`) or `X` and `objective` (`BinaryClassificationPipeline`, `MulticlassClassificationPipeline`) or `X` (`ClassificationPipeline`).

I think adding this extra check for the number of arguments in `predict` is the lowest-touch way of handling the complexity (as opposed to adding another metaclass) especially since this class is not user-facing and we don't expect users to define their own pipeline base classes. 

That being said, maybe we can refactor this metaclass definition to make it a little clearer what's going on?

Reply:
I think checking # of arguments is a good idea. Maybe we can just refactor it into the usual if else logic? 

```
             if not self._is_fitted:
                raise PipelineNotYetFittedError(f'This {klass} is not fitted yet. You must fit {klass} before calling {method.__name__}.')
            elif X is None and y is None:
                return method(self)
            elif y is None:
                return method(self, X)
            # For time series classification pipelines, predict will take X, y, objective
            elif len(inspect.getfullargspec(method).args) == 4:
                 return method(self, X, y, objective)
            else:
                # For other pipelines, predict will take X, y or X, objective
                return method(self, X, y)

Reply:
Done!